cirq/ops/common_channels.py

```diff
@@ -184,6 +195,700 @@ def depolarize(p: float) -> DepolarizingChannel:
+                  + M_2 \rho M_2^\dagger + M_3 \rho M_3^\dagger
+
+        With
+            M_0 = \sqrt{p} \begin{bmatrix}
+                                1 & 0  \\
+                                0 & \sqrt{1 - \gamma}
+                           \end{bmatrix}
+            M_1 = \sqrt{p} \begin{bmatrix}
+                                0 & \sqrt{\gamma} \\
+                                0 & 0
+                           \end{bmatrix}
+            M_2 = \sqrt{1-p} \begin{bmatrix}
+                                \sqrt{1-\gamma} & 0 \\
+                                 0 & 1
+                              \end{bmatrix}
+            M_3 = \sqrt{1-p} \begin{bmatrix}
+                                 0 & 0 \\
+                                 \sqrt{gamma} & 0
+                             \end{bmatrix}
+
+        Args:
+            gamma: The damping rate
+            p: the probability of damping occuring
+
+        Raises:
+            ValueError: if gamma or p is not a valid probability.
+        """
+
+        def validate_probability(p, p_str):
+            if p < 0:
+                raise ValueError('{} was less than 0.'.format(p_str))
+            elif p > 1:
+                raise ValueError('{} was greater than 1.'.format(p_str))
+            return p
+
+        self._gamma = validate_probability(gamma, 'gamma')
+        self._p = validate_probability(p, 'p')
+
+    def _channel_(self) -> Iterable[np.ndarray]:
+        p0 = np.sqrt(self._p)
+        p1 = np.sqrt(1. - self._p)
+        sqrt_g = np.sqrt(self._gamma)
+        sqrt_g1 = np.sqrt(1. - self._gamma)
+        return (
+            p0 * np.array([[1., 0.], [0., sqrt_g1]]),
+            p0 * np.array([[0., sqrt_g], [0., 0.]]),
+            p1 * np.array([[sqrt_g1, 0.], [0., 1.]]),
+            p1 * np.array([[0., 0.], [sqrt_g, 0.]]),
+        )
+
+    def _eq_tuple(self):
+        return (GeneralizedAmplitudeDampingChannel, self._p, self._gamma)
+
+    def __eq__(self, other):
+        if not isinstance(other, type(self)):
+            return NotImplemented
+        return self._eq_tuple() == other._eq_tuple()
+
+    def __hash__(self):
+        return hash(self._eq_tuple())
+
+    def __ne__(self, other):
+        return not self == other
+
+    def __repr__(self) -> str:
+        return 'cirq.generalized_amplitude_damp(p={!r},gamma={!r})'.format(
+            self._p, self._gamma
+        )
+
+    def __str__(self) -> str:
+        return 'generalized_amplitude_damp(p={!r},gamma={!r})'.format(
+            self._p, self._gamma
+        )
+
+    def _circuit_diagram_info_(
+        self, args: protocols.CircuitDiagramInfoArgs
+    ) -> str:
+        return 'GAD({!r},{!r})'.format(self._p, self._gamma)
+
+
+def generalized_amplitude_damp(
+    p: float, gamma: float
+) -> GeneralizedAmplitudeDampingChannel:
+    """
+    Returns a GeneralizedAmplitudeDampingChannel with the given damping rate
+    occuring with the given probability
+
+    This channel evolves a density matrix via
```

Comment:
this doc doesn't match the definition (this is amplitude_damp)

Reply:
fixed.